tplus/evm/contracts.py

```diff
@@ -326,6 +327,47 @@ def __getattr__(self, attr_name: str):
     def from_chain_address(cls, chain_address: ChainAddress) -> "DepositVault":
         return cls(chain_address.chain_id, chain_address.evm_address)
 
+    def deposit(
+        self, user: UserPublicKey, token: AddressType, amount: int, **tx_kwargs
+    ) -> "ReceiptAPI":
+        try:
+            return self.contract.deposit(user, token, amount, **tx_kwargs)
+        except ContractLogicError as err:
+            if erc20_err_name := _decode_erc20_error(err.message):
+                raise ContractLogicError(erc20_err_name) from err
+
+            raise  # Error as-is.
+
+    def execute_atomic_settlement(
+        self,
+        settlement: dict,
+        user: UserPublicKey,
+        data: HexBytes,
+        signature: HexBytes,
+        **tx_kwargs,
+    ) -> "ReceiptAPI":
+        try:
+            return self.contract.executeAtomicSettlement(
+                settlement, user, data, signature, **tx_kwargs
+            )
+        except ContractLogicError as err:
+            if erc20_err_name := _decode_erc20_error(err.message):
+                raise ContractLogicError(erc20_err_name) from err
+
+            raise  # Error as-is.
+
+
+def _decode_erc20_error(err: str) -> ContractLogicError | None:
+    if err == "0x7939f424":
+        # The sender likely didn't approve, or they don't have the tokens.
+        return "TransferFromFailed()"
```

Comment:
Useful helper, but looks like the type annotation is off, string vs ContractLogicError? you know ape better than I do though

Reply:
fixed in this PR: https://github.com/tpluslabs/tpluspy/pull/78